From: hololeap <hololeap@protonmail.com>
Date: Tue, 5 Mar 2024 20:57:17 -0700
Subject: [PATCH 1/1] Fix unit test imports

Do not import shrinkBoundedEnum from Test.QuickCheck, which was added
in QuickCheck-2.14.3, as it collides with a function in the module.

See: https://github.com/nick8325/quickcheck/commit/2982eaf7fab145b1d8156375a4a22ba5e13c370e
Signed-off-by: hololeap <hololeap@protonmail.com>
--- a/tests/UnitTests/Distribution/Client/ArbitraryInstances.hs
+++ b/tests/UnitTests/Distribution/Client/ArbitraryInstances.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE GADTs            #-}
 {-# LANGUAGE TypeOperators    #-}
@@ -44,6 +45,9 @@ import Distribution.Solver.Types.PackageConstraint       (PackageProperty (..))
 import Data.Coerce                      (Coercible, coerce)
 import Network.URI                      (URI (..), URIAuth (..), isUnreserved)
 import Test.QuickCheck
+#if MIN_VERSION_QuickCheck(2,14,3)
+    hiding (shrinkBoundedEnum)
+#endif
 import Test.QuickCheck.GenericArbitrary
 import Test.QuickCheck.Instances.Cabal ()
 
-- 
2.43.0

